### PR TITLE
Site editor sidebar: home template details

### PR DESCRIPTION
**Pull request by ramonjd**
## What?

Displays template and settings configuration details for the home and index templates.

See the designs and specs over at:
- https://github.com/WordPress/gutenberg/issues/49597

## Status

- Rest API changes for template/template part last modified in https://github.com/WordPress/gutenberg/pull/51362
- Clicking on a template area loads that template, but it breaks the back button. This is an existing issue, e.g., clicking on a template in the pages panel, then hitting back will take you to the templates panel (and not back to the pages panel)

## Testing Instructions
With a block theme activated, head over to the site editor and open the home or index template. 
Check that the settings controls work, and that the template area links open the correct template:
1. The posts per page and allow block comments controls, when saved, should be synched with the settings on `/wp-admin/options-reading.php`
2. When a static page is selected for the posts page on `/wp-admin/options-reading.php`, the post title control should display. Updating this field and saving should set the title of the posts page only.

## Screenshots or screencast 

State of this PR:

 | Home template as posts page  | Home template as Homepage |
| ------------- | ------------- |
| Screenshot 2023-06-16 at 11 13 14 am  | Screenshot 2023-06-16 at 11 13 29 am |

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+		const { getEntityRecord } = select( coreStore );
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		const { getSettings } = select( blockEditorStore );
+		const blockEditorSettings = getSettings();
+
+		return {
+			isHomePageBlog:
+				siteSettings?.page_for_posts === siteSettings?.page_on_front,
+			siteTitle: siteSettings?.title,
+			postsPerPage: +siteSettings?.posts_per_page,
```
Comment (andrewserong) [inline packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js]: Just a drive-by comment: what does the plus here do? Since optional chaining is used, what happens if siteSettings or `posts_per_age` is undefined, but the plus is applied beforehand... would that result in NaN?

Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js]: It coerces to a number, but there is a risk it will result in a `NaN`. I was counting on a `null` value, but didn't think much of it to be honest! I'll remove since it's a bit too clever. 

Thanks for flagging it 👍

File: lib/compat/wordpress-6.3/block-template-utils.php [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+ * Builds a unified template object based a post Object.
+ *
+ * @since 5.9.0
+ * @since 6.3.0 Adding template `modified` property.
+ * @access private
+ *
+ * @param WP_Post $post Template post.
+ * @return WP_Block_Template|WP_Error Template or error object.
+ */
+function _gutenberg_build_block_template_result_from_post( $post ) {
```
Comment (ramonjd) [inline lib/compat/wordpress-6.3/block-template-utils.php]: I'm planning to split adding the modified property to the templates response to another PR.

Here just for testing purposes.

**Comment by jameskoster** [comment]
> Discussion and comments controls/settings are not hooked up

~There was some conversation in the issue about whether to include these on blog templates or not. The concern was that their inclusion would suggest the setting is blog specific, when in fact it applies to all post types. The same can be said for the Posts per page setting.~

Edit: Probably fine to include, if we append some help text explaining the global nature of the settings:

Screenshot 2023-06-07 at 12 36 09



---

I know it's still a WIP, but I just wanted to note that in the Areas section, each entire row should act as a button, rather than only the name of the template part alone. Mockup:

Screenshot 2023-06-07 at 11 13 31

---

I suspect it might be worth cc'ing @ciampo and @mirka for any insight on how best to theme the input components.

**Comment by ciampo** [comment]
> I suspect it might be worth cc'ing @ciampo and @mirka for any insight on how best to theme the input components.

Thank you for the ping!

While the `Theme` component is not ready to be used yet, we will have to rely on style overrides. The best thing would be to group all of the style overrides (text, background and border color) in one place/file, so that they can be easily found and removed once theming can be used. We could also add a comment wrapping those styles, stating that they are a temporary solution and that in general style overrides are discouraged.

**Comment by ramonjd** [comment]
Thanks for the updated designs @jameskoster !

>  The best thing would be to group all of the style overrides (text, background and border color) in one place/file, so that they can be easily found and removed once theming can be used. We could also add a comment wrapping those styles, stating that they are a temporary solution and that in general style overrides are discouraged.

Good idea, will do. 👍

**Comment by ramonjd** [comment]
I've removed "Last updated" changes because the current API does not support it. 

We can reinstate once this PR is merged (if it does):
- https://github.com/WordPress/gutenberg/pull/51362

File: packages/edit-site/src/components/sidebar-navigation-screen-template/index.js [hunk 3]
```diff
@@ -42,6 +43,11 @@ function useTemplateTitleAndDescription( postType, postId ) {
 		);
 	}
 
+	let content = null;
+	if ( record?.slug === 'home' || record?.slug === 'index' ) {
```
Comment (tellthemachines) [inline packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: If this applies to all blog templates, shouldn't it include the Archives template too? Or should it only apply to Home, with Index as a fallback if Home doesn't exist? Settings such as "site title" only really make sense on Home I guess 🤔

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+function TemplateAreaButton( { area, postId, icon, title } ) {
+	const icons = {
+		header,
+		footer,
+	};
+	// Generic area labels - not the same as the template part title defined in "templateParts" in theme.json.
+	const areaLabels = {
+		header: __( 'Header' ),
+		footer: __( 'Footer' ),
+	};
```
Comment (tellthemachines) [inline packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js]: Are there only ever these two area labels, or is this extensible by themes at all?

File: packages/edit-site/src/components/sidebar-navigation-screen/style.scss [hunk 5]
```diff
@@ -99,3 +99,24 @@
...
 	padding: $grid-unit-20 0;
 }
+
+/* In general style overrides are discouraged.
+ * This is a temporary solution to override the InputControl component's styles.
+ * The `Theme` component will potentially be the more appropriate approach
+ * once that component is stabilized.
+ * See: packages/components/src/theme
+ */
+.edit-site-sidebar-navigation-screen__input-control {
```
Comment (tellthemachines) [inline packages/edit-site/src/components/sidebar-navigation-screen/style.scss:106]: Could we add a temporary override for the help text color too? It's really low contrast against the dark background and barely legible.

**Review by tellthemachines — COMMENTED** [review]
This is testing really well! Just a couple questions below (mostly me trying to understand this better)

File: packages/edit-site/src/components/sidebar-navigation-screen-template/index.js (hunk 3 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: Good question! I wasn't really sure about this. Happy to take instructions from @jameskoster here 😄

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js (hunk 4 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js]: The design looks like this:



I'm guessing it's a generic label for the template part, at least the "official" ones that can be in a template. 

So in that sense, it's incomplete.

But we already have the `title` (displayed on the right), which comes from theme.json (if it's set), so it seems redundant (?)

Comment (jameskoster) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js]: The label on the left was intended to be the semantic area which can be useful when the template part name doesn't provide that information. That said we can probably make do with the icon and omit the label if it helps:

Screenshot 2023-06-13 at 11 23 32

File: packages/edit-site/src/components/sidebar-navigation-screen-template/index.js (hunk 3 above)
Comment (jameskoster) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: No, because Archive and so on can display custom post types too. It may be that we add a local posts-per-page setting to to templates like Archive in the future, but let's start with the global stuff for now.

> Settings such as "site title" only really make sense on Home I guess

The setting here should be Blog title, not Site title. And it should only be visible when a Posts Page is set. Updating the Blog title simply updates the title of the Posts Page.

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js (hunk 4 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js]: Thanks for the updated design! Done.

File: packages/edit-site/src/components/sidebar-navigation-screen-template/index.js (hunk 3 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: > Updating the Blog title simply updates the title of the Posts Page.

@jameskoster Ah, that was a misinterpretation on my part, sorry!

I've updated this PR so that the "Blog title" input control only shows when a static page is selected as the posts page. The value of the input control updates the title of that page.

Screenshot 2023-06-14 at 2 09 49 pm

Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: Just reflecting on the above change, I think I was unclear about why we would want to update the title of the posts page from the home template, when the two could have no relationship. For example, if I select page `X` as my posts page, it might use a page template and not the home or index templates.

Unless the intention is to surface general site settings?

The only other conclusion I could reach is we'd show these controls when we set a static page as the posts page and _**that**_ static page uses the home/index template. 

Anyway, let me know if I'm way off.

Comment (jameskoster) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: > For example, if I select page X as my posts page, it might use a page template and not the home or index templates.

Please feel free to double-check but I think the Posts page will _always_ resolve to either Home or Index. Note that the template switching UI in the post editor does not allow you to change the template when editing this page;

Screenshot 2023-06-14 at 11 05 59

> I've updated this PR so that the "Blog title" input control only shows when a static page is selected as the posts page. The value of the input control updates the title of that page.

If the above is true, this sounds perfect!

**Comment by jameskoster** [comment]
This is looking pretty good, thanks for all the hard work @ramonjd :)

So, I think I have uncovered a bit of an issue in updating the _global_ posts per page setting in this panel. Many themes (including twenty twenty-three) set Query Loop blocks to `inherit`, but also override properties like `perPage`. [Relevant code](https://github.com/WordPress/twentytwentythree/blob/11440a5b38bae6882fc2c00b703d4bb95420a0f5/templates/home.html#L9). 

Consequently, updating the global posts per page setting here won't _always_ update how many posts per page are displayed on this template, and that is going to make for a confusing experience 🙃 

I'm very open to ideas here, but (if possible) one option could be to link the Posts per page control in the sidebar to the Query Loop in the template. There would be the following potential scenarios:

* Query Loop has no `perPage` value: Control displays global value as a `placeholder`. Adding a value would add the property to the Query Loop.
* Query Loop has a `perPage` value: Control displays that value. Changing it updates the Query Loop. Removing it would remove the property from Query Loop, and the control would display the global value.

Here's a mockup:

Screenshot 2023-06-14 at 12 27 27

What do you think?

---

A couple of smaller details:

* One of the main benefits of having the settings and the frame visible together is that you can see the affect that changing settings will have in real time. With that said, could the frame update when you change the posts per page setting?
* I suspect this will be best handled in a follow-up, but it's quite important: blocks like Post Title in the Home template should reflect the associated page data. Notice here how the Blog title input and the Post Title block are displaying different information:

Screenshot 2023-06-14 at 11 27 39

**Comment by ramonjd** [comment]
I think I get why I was stumbling over the requirements, thanks for helping me understand @jameskoster 

I think my main problem was trying make the cognitive link between the specific home page template and overall site settings. For example,  "Allow comments on new posts" doesn't have any direct relationship to the home page template being edited/viewed. 

I'm wondering if these things will cause more confusion unless we better communicate their intended effect.

> one option could be to link the Posts per page control in the sidebar to the Query Loop in the template

So, when I open a homepage template in view mode, we'd want to control (and update reactively) query block attributes in the editor from the sidebar. Or in other words, update block attributes in real time from the site editor sidebar in "view" mode.

It might be possible to drill down the template's block tree and find the first query block to display how many posts a home page query block displays. Even, use the `clientId` of a block and update its attributes via `updateBlockAttributes` (I haven't tried either of these yet). I would lean away from this idea: it's possible to have more than one query block (or any block) in a template, and it might be computationally heavy.

I'm just wondering why we couldn't guide folks to open the editor and do this in "edit" mode? 

It's what the editor is set up for and what it does best. 

> I suspect this will be best handled in a follow-up, but it's quite important: blocks like Post Title in the Home template should reflect the associated page data. Notice here how the Blog title input and the Post Title block are displaying different information:

I think this would only make sense to do on the individual page, and only when that page is set to display posts. The post title block appears only on the page template that is set as the posts page in the reading settings.

So, taking 2023's home page template as an example, the "Mindblown: a blog about philosophy." heading is just a heading block. 

Screenshot 2023-06-15 at 10 04 13 am

Updating it would require first finding, then updating that particular heading block's content. There could be 1 heading there could be 100 on a template and we wouldn't really know which heading acts as the "title".

Plus, and this is just my personal opinion, to me it blurs the lines of "view" and "edit" modes to allow updating of block-specific attributes from the sidebar. 

---- 

This close to code freeze for 6.3, my vote would be to revisit this functionality. There's the risk of introducing something that half works or is buggy, if we can get things working at all. 

With all this in mind, and at this stage, I think we could display some info that we know about the home template, e.g., like the page details panel. For example, we can already display template areas that appear on the template, e.g., footer and header (already done)

Furthermore, if a "home" template exists, and the reading settings are set to "homepage displays displays your latest posts", then we could display this.

I'll try to play with a few things so we have options.

**Comment by ramonjd** [comment]
Without the controls, here's something we could do now:

Screenshot 2023-06-15 at 4 11 13 pm

The last updated footer should appear on all template and template part panels once https://github.com/WordPress/gutenberg/pull/51362 is merged.

The "This template is currently set as your homepage" label will only show if:

- Your homepage displays your latest posts (in the reading settings)
- There's no `front-page` template (which overrides any home page)

**Comment by jameskoster** [comment]
> For example, "Allow comments on new posts" doesn't have any direct relationship to the home page template being edited/viewed

It makes sense because the Home template is the blog home. It will always display latest posts whether that is as the site homepage, or the posts page. So it seems reasonable to find settings relating to the blog there. 

> it's possible to have more than one query block (or any block) in a template, and it might be computationally heavy

Yes I thought this might be a bit tricky. I don't know if it helps, but we'd be looking for the Query Loop with `"inherit":true`, and it seems unlikely that there would be more than one of those. 

> I'm just wondering why we couldn't guide folks to open the editor and do this in "edit" mode?

This isn't an option unfortunately, because there is no UI to edit properties like `perPage` when the Query Loop is`"inherit":true`.

> I think this would only make sense to do on the individual page, and only when that page is set to display posts. The post title block appears only on the page template that is set as the posts page in the reading settings.

The thing is, the page and the template are basically the same thing because the posts 'page' will _always_ resolve to display the Home (or Index) template. So for this particular case (posts page) it's the Home template that includes the Post Title block. That's why it makes sense for it to be editable while viewing the template. 

> So, taking 2023's home page template as an example, the "Mindblown: a blog about philosophy." heading is just a heading block.

To be clear, I'm not suggesting we should update that "Mindblown..." heading, or any other random Heading block. The "Blog title" input in the sidebar should update the title of the Posts Page. Consequently any instance of the Post Title block in the Home template would be inherently updated automatically. This was working correctly when I last built the PR. The piece that was missing was the frame updating without having to reload the page. 

Hopefully this extra context helps narrow the scope a bit. If you think it might be worth looking into separately that's probably fine. It would be nice to merge _something_ in time for 6.3 because the current template detail panels are rather lacklustre. Even if it's just the links to the template parts and the revisions.

> Without the controls, here's something we could do now:

I don't think the Theme and Slug details are super useful. For the "This template is your homepage" text, we actually have a dedicated issue for that one (https://github.com/WordPress/gutenberg/issues/50582), so it would be good to handle separately.

**Comment by jameskoster** [comment]
I thought about the Posts per page setting some more and came to the conclusion that we should stick to the original plan. That is: include the control as a way to update the global `posts_per_page` value. The main reason being that the TT3 issue _already exists_ albeit somewhat buried — if you edit `posts_per_page` in reading settings, the TT3 Home template will ignore it. This is a separate issue that should probably be addressed somehow on the Query Loop block. Edit: I opened https://github.com/WordPress/gutenberg/issues/51535 to discuss, but it's not high priority. 

In the short term we can include some help text to better explain why the setting might not be applied:

Home template

In short:

* Include the Blog Title control, which updates the Posts Page title. Do not include if no Posts Page is set.
* Include posts per page control, which updates the global posts per page reading setting.
* Include the discussion setting.

We can live without the dynamic preview updating if that's a big headache. There would hopefully be time to wrangle that later.

File: packages/edit-site/src/components/sidebar-navigation-screen-template/index.js (hunk 3 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/index.js:47]: > I think the Posts page will always resolve to either Home or Index. 

Oh yes, good one! Another piece of the puzzle falls into place. 😄 

>  I've updated this PR so that the "Blog title" input control only shows when a static page is selected as the posts page. The value of the input control updates the title of that page.

Just to confirm, and maybe you're aware of this already. Here's what I'm seeing:

1. Select a page to act as the posts page (it uses the home page template in 2022 and 2023)
Screenshot 2023-06-14 at 2 09 49 pm

2. Here's the page in question. The title is highlighted
Screenshot 2023-06-16 at 10 38 04 am

3. When updating the post page title, the only thing that will change (only after save) on the home page is the nav menu item.

Screenshot 2023-06-16 at 10 48 52 am

**Comment by ramonjd** [comment]
> I thought about the Posts per page setting some more and came to the conclusion that we should stick to the original plan.

No worries.

For home and index templates:

✅ Include the Blog Title control, which updates the Posts Page title. Do not include if no Posts Page is set.
✅ Include posts per page control, which updates the global posts per page reading setting.
✅ Include the discussion setting.

> We can live without the dynamic preview updating if that's a big headache

Until we come up with an acceptable way to locate and update the right blocks in a performant way, I'd probably leave these at this late stage. 👍 

 Here's the state of the branch:

 | Home template as posts page  | Home template as Homepage |
| ------------- | ------------- |
| Screenshot 2023-06-16 at 11 13 14 am  | Screenshot 2023-06-16 at 11 13 29 am |

**Comment by jameskoster** [comment]
Thanks for the updates. 

After some testing I realised that adding a Post Title block to the Home template won't actually retrieve and display the Posts Page title 🤦‍♂️. I think there's still some value in including the setting, I'll just update the help text to explain where it is used. 

A couple of small changes and I think we're good to go:

* For the template part buttons, could we use `NavigatorButton` for consistency with other such items in the sidebar? Hopefully that will also fix the current alignment / spacing issues.
* Possibly not caused by this PR and a bug to fix separately: the Home template doesn't display the correct number of posts in the editor. The frontend is fine.

**Comment by ramonjd** [comment]
Thanks @jameskoster !!

> For the template part buttons, could we use NavigatorButton for consistency with other such items in the sidebar? Hopefully that will also fix the current alignment / spacing issues.

I tried this, but `<NavigatorButton / >` is specifically built for menu navigation and requires a `path`, which doesn't match the URL pattern for individual entities. I suppose it's the same reason why the pages view doesn't use it. There is `<SidebarNavigationItem />` however, which the pages view does use. I'll try to use that one since it's styled already. 🤞 


> Possibly not caused by this PR and a bug to fix separately: the Home template doesn't display the correct number of posts in the editor. The frontend is fine.

As far as I can tell, this is caused by the Query template block edit JS, which doesn't appear to update the query when the `postsPerPage` value doesn't match the `perPage` attribute value in this [hook](https://github.com/WordPress/gutenberg/blob/e41b94550e7e3636e378a664b8cbde0149a50efb/packages/block-library/src/query/edit/query-content.js#L64).

cc @ntsekouras to check if that's intended or a bug. I'm happy to get an issue up if the latter.

**Comment by ramonjd** [comment]
> There is <SidebarNavigationItem /> however, which the pages view does use. I'll try to use that one since it's styled already.


Looks good (I think) 😄

**Comment by ntsekouras** [comment]
>As far as I can tell, this is caused by the Query template block edit JS, which doesn't appear to update the query when the postsPerPage value doesn't match the perPage attribute value in this [hook](https://github.com/WordPress/gutenberg/blob/e41b94550e7e3636e378a664b8cbde0149a50efb/packages/block-library/src/query/edit/query-content.js#L64).

cc @ntsekouras to check if that's intended or a bug. I'm happy to get an issue up if the latter.

I created a PR for the proper init/update of reading setting [here](https://github.com/WordPress/gutenberg/pull/51641).

**Comment by jameskoster** [comment]
> Looks good (I think) 😄

The menu items themselves look good 🎉 The alignment was a bit off though. To add the negative margin and fix the alignment I wrapped them in an `ItemGroup`, which is the same method used on the Pages / Templates panel. If there's a better way please feel free to revert and fix :D 

I also tweaked a couple other of details to match some metrics to the latest design developments.

I reckon this is good to go now, but would suggest a couple of follow-ups:

1. I tried to use "custom" `spinControls` on the `NumberControl`, but it felt very laggy for some reason
2. We should iterate on the author detail that appears when the template is user-generated, but this needs a little design thought:

Screenshot 2023-06-19 at 12 31 05

**Review by jameskoster — APPROVED** [review]

**Comment by ramonjd** [comment]
Thanks a lot @jameskoster! 🙇 

> I tried to use "custom" spinControls on the NumberControl, but it felt very laggy for some reason

That would be the `debouce()` I can remove it since we're only stashing changes in the store and not saving.

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+			? templateRecord.blocks
+					.filter( ( { name } ) => name === 'core/template-part' )
+					.map( ( { attributes } ) => ( {
+						...templatePartAreas?.find(
+							( { area } ) => area === attributes?.tagName
+						),
+						...attributes,
+					} ) )
+			: [];
+	}, [ templateRecord?.blocks, templatePartAreas ] );
```
Comment (andrewserong) [inline packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js:132]: This currently only displays template part areas that are at the root of the document, so for a site that has nested the header or footer template parts within a Group block, they don't appear to show up. In the following screenshot, the site's header is wrapper in a Group block, but the Footer isn't, so the Footer is showing but not the Header:

image

Would it be worth re-using the [`getFilteredTemplatePartBlocks` utility function](https://github.com/WordPress/gutenberg/blob/a065bb5263514f669b1662da9790557980e0f952/packages/edit-site/src/store/utils.js#L69) we use in the right hand template sidebar within the site editor, to iterate into nested template part areas?

This is a bit of a nit, and perfectly fine to address in a follow-up if you'd prefer, since this PR is already pretty big.

Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js:132]: > Would it be worth re-using the [getFilteredTemplatePartBlocks utility function](https://github.com/WordPress/gutenberg/blob/a065bb5263514f669b1662da9790557980e0f952/packages/edit-site/src/store/utils.js#L69) we use in the right hand template sidebar within the site editor, to iterate into nested template part areas?

If it's already in there and tested, then I think it's a low risk. 👍

Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js:132]: Actually, I might add it in a follow up. I wanted to add the last updated footer to template parts as well 🤔 

You're right, the PR is big enough 😄

File: lib/compat/wordpress-6.3/class-gutenberg-rest-templates-controller-6-3.php [hunk 7]
```diff
@@ -100,7 +100,7 @@ public function prepare_item_for_response( $item, $request ) { // phpcs:ignore V
 	/**
 	 * Adds revisions to links.
 	 *
-	 * @since 6.3.0
+	 * @since 6.2.0
```
Comment (andrewserong) [inline lib/compat/wordpress-6.3/class-gutenberg-rest-templates-controller-6-3.php]: Nit: It seems that the revision links code never made it in 6.2.0 from the sounds of it: https://github.com/WordPress/gutenberg/pull/45215#issuecomment-1592704026, so should this still be `6.3.0`?

File: packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+ * Internal dependencies
+ */
+import {
+	SidebarNavigationScreenDetailsPanelRow,
+	SidebarNavigationScreenDetailsPanelLabel,
+	SidebarNavigationScreenDetailsPanelValue,
+} from '../sidebar-navigation-screen-details-panel';
+
+export default function SidebarNavigationScreenDetailsFooter( {
+	lastModifiedDateTime,
```
Comment (andrewserong) [inline packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js:18]: Tiny nit: again not-blocking, but since this is a generic footer component, I was wondering if the check for whether or not to render the last modified row should exist within this component instead?

Just a thought though, and I wouldn't worry about changing it in this PR.

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+	const [ commentsOnNewPostsValue, setCommentsOnNewPostsValue ] =
+		useState( '' );
+	const [ postsCountValue, setPostsCountValue ] = useState( 1 );
+	const [ postsPageTitleValue, setPostsPageTitleValue ] = useState( '' );
+
+	useEffect( () => {
+		setCommentsOnNewPostsValue( allowCommentsOnNewPosts );
+		setPostsPageTitleValue( postsPageTitle );
+		setPostsCountValue( postsPerPage );
+	}, [ postsPageTitle, allowCommentsOnNewPosts, postsPerPage ] );
```
Comment (andrewserong) [inline packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js:119]: This syncs the server-retrieved values to local state, yeah? Only if you're making further changes anyway, would it be worth adding a comment just before this useEffect? It took me a second to work out what was happening.

**Review by andrewserong — APPROVED** [review]
Nice work @ramonjd, this is looking good to me!

The features were all testing well according to the designs and how it's been developed 👍

The main issue I ran into is that the `Blog title` control feels a little edge-casey to me, in that when a posts page is set to the home page, the post title block doesn't appear to output anything within that template as far as I could tell. So, when using it, I didn't feel like I was updating the blog title, and I found it hard to draw the connection between this input and updating the title for the page itself. I see that the help text for the field explains that the title will appear in search results and when the page is shared on social media, but in terms of the UI, I found it difficult to understand how the feature is meant to be used. Since we're editing the Home template, as a user, my expectation is that Blog Title actually means Site Title. That could just be me, though!

The discussion surrounding that control could very well become a bit rabbit-holey, so I reckon it'd be a good idea to merge this PR in its current shape, and defer further exploration of the Blog Title control to follow-ups.

LGTM! ✨

File: lib/compat/wordpress-6.3/class-gutenberg-rest-templates-controller-6-3.php (hunk 7 above)
Comment (ramonjd) [reply lib/compat/wordpress-6.3/class-gutenberg-rest-templates-controller-6-3.php]: I have no idea what that is doing there. Thanks for flagging it. 
Brain meltdown.

Comment (ramonjd) [reply lib/compat/wordpress-6.3/class-gutenberg-rest-templates-controller-6-3.php]: Oh, bad rebase from https://github.com/WordPress/gutenberg/pull/51362/files#diff-4592dfe24434861ba8dc2c1dbd3b6c83528b1e62050e9bb834e7d9f256a83058L101

File: packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js (hunk 8 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-details-footer/index.js:18]: Excellent. I'll add it to the imminent follow-up

File: packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js (hunk 9 above)
Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-template/home-template-details.js:119]: It would be worth it. Added to follow up notes. Thank you!

File: packages/edit-site/src/components/sidebar-navigation-screen-page/page-details.js [hunk 10]
```diff
@@ -138,26 +139,21 @@ export default function PageDetails( { id } ) {
 		[ record?.parent ]
 	);
 	return (
-		<VStack spacing={ 5 }>
```
Comment (andrewserong) [inline packages/edit-site/src/components/sidebar-navigation-screen-page/page-details.js:141]: Apologies, I missed this earlier, but I think the removal of `spacing` here caused a regression for the Details section in the Pages screen:

| Before | After |
| --- | --- |
| image | image |

Could we add back in spacing (or the ability to pass down a value for it to `SidebarNavigationScreenDetailsPanel` in the follow-up PR in https://github.com/WordPress/gutenberg/pull/51669? Edit: it already accepts it, so it might just be a matter of adding in `spacing` on this line: https://github.com/WordPress/gutenberg/blob/7efbf84d43bd033cf689dd11f16177ce98ff87ba/packages/edit-site/src/components/sidebar-navigation-screen-page/page-details.js#L142?

Comment (ramonjd) [reply packages/edit-site/src/components/sidebar-navigation-screen-page/page-details.js:141]: Good spotting. Done.
